File: app/src/main/java/com/tbear9/openfarm/api/UserVariable.java
package com.tbear9.openfarm.api;

import lombok.Builder;
import lombok.Getter;
import lombok.Singular;

import java.io.Serial;
import java.io.Serializable;
import java.util.Set;

@Getter
@Builder
public class UserVariable implements Serializable {
    @Serial private static final long serialVersionUID = 1L;
    private byte[] image;
    private final String tanah;
    @Singular private final Set<? extends Parameters> parameters;
}

File: app/src/main/java/com/trbear9/plants/api/UserVariable.kt
package com.trbear9.plants.api

import android.content.Context
import android.graphics.Bitmap
import lombok.*
import java.nio.charset.StandardCharsets
import java.security.MessageDigest
import java.util.*

@Getter
@AllArgsConstructor
@NoArgsConstructor
class UserVariable {
    @Setter
    private var tanah: String? = null

    //parameters
    var soil: SoilParameters = SoilParameters()
    var geo: GeoParameters = GeoParameters()
    var custom: CustomParameters? = null

    var image: Bitmap? = null
    var context: Context? = null
    var hash: String? = null
}
